src/mds/FSMap.h

```diff
@@ -447,6 +447,29 @@ class FSMap {
     fs->mds_map.epoch = epoch;
   }
 
+  void rename_fs(fs_cluster_id_t fscid, std::string& new_fs_name)
+  {
+    auto& fs = filesystems.at(fscid);
+    fs->mds_map.set_fs_name(new_fs_name);
+    fs->mds_map.epoch = epoch;
+  }
+
+  void swap_fscids(fs_cluster_id_t fscid1, fs_cluster_id_t fscid2)
+  {
+    auto fs1 = filesystems.at(fscid1);
+    auto fs2 = filesystems.at(fscid2);
+    filesystems[fscid1] = fs2;
+    filesystems[fscid2] = fs1;
+    for (auto it : mds_roles) {
+      if (it.second == fscid1)
+	mds_roles[it.first] = fscid2;
+      if (it.second == fscid2)
+	mds_roles[it.first] = fscid1;
+    }
```

Comment:
Hmm, this terrifies me. @ajarr do you remember if we audited how the MDS may blow up if the file system name changes in an MDSMap update? This takes it a step further and changes the fscid!

I think we might add a restriction that no MDS is `up` when doing a rename/swap. i.e. the operator must do:

```
ceph fs fail <name>
ceph fs swap | ceph fs rename
ceph fs set <name> joinable true
```

Reply:
cc @vshankar

Reply:
@rishabh-d-dave while you're at it, please fix `fs rename` too. Maybe separate tracker/PR.

Reply:
True, I'll get it done.